framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -1238,43 +1260,30 @@ public Void visitMethodInvocation(MethodInvocationTree node, Void p) {
 
     /**
      * Checks that the following rule is satisfied: The type on a constructor declaration must be a
-     * supertype of the return type of "super()" invocation within that constructor (except for the
-     * Initialization and Rawness Checkers).
+     * supertype of the return type of "super()" invocation within that constructor.
      */
-    void checkSuperConstructorCall(MethodInvocationTree node) {
-        if (!TreeUtils.isSuperCall(node)) {
-            return;
-        }
-        TreePath path = atypeFactory.getPath(node);
+    protected void checkSuperConstructorCall(MethodInvocationTree superCall) {
```

Comment:
Do we need something similar for a `this` constructor call? The result types could also be inconsistent there, right?

Reply:
Yes! I implemented it.